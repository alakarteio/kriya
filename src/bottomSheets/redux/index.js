import bottomSheets from './bottomSheets'
import * as actions from './bottomSheets.actions'
import * as selectors from './bottomSheets.selectors'

Object.assign(bottomSheets, actions, selectors)

export * from './bottomSheets.actions'
export * from './bottomSheets.selectors'
export default bottomSheets
