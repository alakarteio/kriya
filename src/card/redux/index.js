export { default } from './card'
export * from './card.actions'
export * from './card.selectors'
