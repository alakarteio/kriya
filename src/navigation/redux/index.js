export { default } from './navigation'
export * from './navigation.actions'
export * from './navigation.selectors'
