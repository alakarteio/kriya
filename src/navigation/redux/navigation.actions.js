export const ADD_STEP = 'NAVIGATION_ADD_STEP'
export const addStep = step => ({ type: ADD_STEP, payload: step })

export const RESET_STEPS = 'NAVIGATION_RESET_STEPS'
export const resetSteps = () => ({ type: RESET_STEPS })

export const CLICKED = 'NAVIGATION_CLICKED'
export const click = number => ({ type: CLICKED, payload: number })
