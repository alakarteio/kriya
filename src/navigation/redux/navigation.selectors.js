export const getNavigation = state => state.ui.navigation
