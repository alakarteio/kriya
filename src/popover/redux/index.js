export { default } from './popover'
export * from './popover.actions'
export * from './popover.selectors'
