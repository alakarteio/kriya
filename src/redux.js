export { default as cards } from './card/redux'
export { default as navigation } from './navigation/redux'
export { default as popover } from './popover/redux'
export { default as bottomSheets } from './bottomSheets/redux'
export { default as tabs } from './tabs/redux'
