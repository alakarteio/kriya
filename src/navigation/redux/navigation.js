import { ADD_STEP, RESET_STEPS } from './navigation.actions'

export const initState = []
export const initAction = { type: 'UNKNOWN' }

export default (state = initState, { type, payload } = initAction) => {
  switch (type) {
    case ADD_STEP: return [...state, payload]
    case RESET_STEPS: return initState
    default: return state
  }
}
