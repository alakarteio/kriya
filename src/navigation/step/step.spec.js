/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import Step from './step'
import StepContainer from './step.container'

const snap = (props) => {
  const component = renderer.create(
    <Step children={['Hello step']} onClick={() => {}} {...props} />, // eslint-disable-line react/no-children-prop
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
}

describe('common/Step', () => {
  describe('graphical (JSX)', () => {
    it('should call onClick when clicked', () => {
      const onClick = jest.fn()

      const wrapper = shallow(
        <Step onClick={onClick}>{'a child for onClick'}</Step>,
      )
      wrapper.find('button').simulate('click')

      expect(onClick.mock.calls.length).toBe(1)
    })

    it('should add custom className', () => snap({ className: 'custom' }))
    it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
    it('should add children', () => snap({ children: ['a child'] }))
    it('should have a default behaviour', () => snap({}))
    it('should not print separator on last item', () => snap({ last: true }))
  })

  describe('container', () => {
    it('should dispatch a navigation event', () => {
      const store = createStore(() => ({}))
      const dispatch = jest.fn()
      store.dispatch = dispatch

      const wrapper = mount(
        <Provider store={store}>
          <StepContainer number={42}>{'text'}</StepContainer>
        </Provider>,
      )

      wrapper.find('button').simulate('click')

      expect(dispatch.mock.calls.length).toBe(1)
      expect(dispatch.mock.calls[0]).toEqual([{ type: 'NAVIGATION_CLICKED', payload: 42 }])
    })
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
