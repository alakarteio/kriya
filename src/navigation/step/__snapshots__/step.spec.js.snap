// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Step graphical (JSX) should add children 1`] = `
<div
  className="step intermediate"
  style={Object {}}
>
  <button
    onClick={[Function]}
  >
    a child
  </button>
  <div
    className="separator"
  >
     /
  </div>
</div>
`;

exports[`common/Step graphical (JSX) should add custom className 1`] = `
<div
  className="step custom intermediate"
  style={Object {}}
>
  <button
    onClick={[Function]}
  >
    Hello step
  </button>
  <div
    className="separator"
  >
     /
  </div>
</div>
`;

exports[`common/Step graphical (JSX) should add custom style 1`] = `
<div
  className="step intermediate"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <button
    onClick={[Function]}
  >
    Hello step
  </button>
  <div
    className="separator"
  >
     /
  </div>
</div>
`;

exports[`common/Step graphical (JSX) should have a default behaviour 1`] = `
<div
  className="step intermediate"
  style={Object {}}
>
  <button
    onClick={[Function]}
  >
    Hello step
  </button>
  <div
    className="separator"
  >
     /
  </div>
</div>
`;

exports[`common/Step graphical (JSX) should not print separator on last item 1`] = `
<div
  className="step last"
  style={Object {}}
>
  <button
    onClick={[Function]}
  >
    Hello step
  </button>
</div>
`;
